src/neo/Network/P2P/TaskManager.cs

```diff
@@ -95,7 +95,7 @@ private void OnNewTasks(InvPayload payload)
             if (!sessions.TryGetValue(Sender, out TaskSession session))
                 return;
             // Do not accept payload of type InventoryType.TX if not synced on best known HeaderHeight
-            if (payload.Type == InventoryType.TX && Blockchain.Singleton.Height < sessions.Values.Max(p => p.LastBlockIndex))
+            if (payload.Type == InventoryType.TX && sessions.Values.Where(p => p.LastBlockIndex > Blockchain.Singleton.Height + 12).Count() > sessions.Count / 3)
                 return;
```

Comment:
I think that it's better to remove this lines, the mempool will work as expected, if it's filled, will be trimmed, so why we need to avoid this messages during sync based on untrusted information?

Reply:
That's the way neo-go currently works (`IsInSync` logic mentioned earlier is only used to determine when we should start dbft). The only problem with this is that the node can try relaying invalid transactions which can be interpreted as suspicious behaviour by its peers. But these transactions should at the same time fall within `MaxValidUntilBlockIncrement` range from current node's height which is not really likely to happen.